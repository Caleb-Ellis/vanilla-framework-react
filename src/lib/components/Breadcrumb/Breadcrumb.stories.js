import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';

import Breadcrumb from './Breadcrumb';
import BreadcrumbItem from './BreadcrumbItem';

storiesOf('Breadcrumb', module)
  .add('Default',
    withInfo('You can use the breadcrumbs pattern to indicate where the current page sits in the sites navigation. The separators between each item are added via CSS, so you dont have to include them manually.')(() => (
      <Breadcrumb>
        <BreadcrumbItem><a href="#a">Breadcrumb One</a></BreadcrumbItem>
        <BreadcrumbItem><a href="#a">Breadcrumb Two</a></BreadcrumbItem>
        <BreadcrumbItem><a href="#a">Breadcrumb Three</a></BreadcrumbItem>
        <BreadcrumbItem>Breadcrumb Four</BreadcrumbItem>
      </Breadcrumb>),
    ),
  );
