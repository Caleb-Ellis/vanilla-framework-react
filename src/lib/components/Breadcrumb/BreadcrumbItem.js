import React from 'react';
import PropTypes from 'prop-types';
import './Breadcrumb.css';

const BreadcrumbItem = props => (
  <li className="p-breadcrumbs__item">
    { props.children }
  </li>
);

BreadcrumbItem.propTypes = {
  children: PropTypes.node.isRequired,
};

export default BreadcrumbItem;
