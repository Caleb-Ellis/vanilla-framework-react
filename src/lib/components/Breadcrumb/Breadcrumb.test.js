import React from 'react';
import ReactTestRenderer from 'react-test-renderer';
import Breadcrumb from './Breadcrumb';
import BreadcrumbItem from './BreadcrumbItem';

describe('Breadcrumb component', () => {
  it('should render a simple Breadcrumb correctly', () => {
    const breadcrumb = ReactTestRenderer.create(
      <Breadcrumb>
        <BreadcrumbItem><a href="#a">Lorem</a></BreadcrumbItem>
        <BreadcrumbItem><a href="#b">Ipsum</a></BreadcrumbItem>
        <BreadcrumbItem><a href="#c">Dolor</a></BreadcrumbItem>
      </Breadcrumb>,
    );
    const json = breadcrumb.toJSON();
    expect(json).toMatchSnapshot();
  });

  it('should render a Breadcrumb without links', () => {
    const breadcrumb = ReactTestRenderer.create(
      <Breadcrumb>
        <BreadcrumbItem><a href="#a">Lorem</a></BreadcrumbItem>
        <BreadcrumbItem><a href="#b">Ipsum</a></BreadcrumbItem>
        <BreadcrumbItem>Dolor</BreadcrumbItem>
      </Breadcrumb>,
    );
    const json = breadcrumb.toJSON();
    expect(json).toMatchSnapshot();
  });
});
