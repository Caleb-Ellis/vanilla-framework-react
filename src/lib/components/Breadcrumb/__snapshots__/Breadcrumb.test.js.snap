// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Breadcrumb component should render a Breadcrumb without links 1`] = `
<ul
  className="p-breadcrumbs"
>
  <li
    className="p-breadcrumbs__item"
  >
    <a
      href="#a"
    >
      Lorem
    </a>
  </li>
  <li
    className="p-breadcrumbs__item"
  >
    <a
      href="#b"
    >
      Ipsum
    </a>
  </li>
  <li
    className="p-breadcrumbs__item"
  >
    Dolor
  </li>
</ul>
`;

exports[`Breadcrumb component should render a simple Breadcrumb correctly 1`] = `
<ul
  className="p-breadcrumbs"
>
  <li
    className="p-breadcrumbs__item"
  >
    <a
      href="#a"
    >
      Lorem
    </a>
  </li>
  <li
    className="p-breadcrumbs__item"
  >
    <a
      href="#b"
    >
      Ipsum
    </a>
  </li>
  <li
    className="p-breadcrumbs__item"
  >
    <a
      href="#c"
    >
      Dolor
    </a>
  </li>
</ul>
`;
